src/service/url-replacements-impl.js

```diff
@@ -445,94 +459,166 @@ export class UrlReplacements {
    * The data for the timing events is retrieved from performance.timing API.
    * If start and end events are both given, the result is the difference between the two.
    * If only start event is given, the result is the timing value at start event.
-   * @param {*} startEvent
-   * @param {*=} endEvent
-   * @return {!Promise<string|undefined>}
+   * @param {string} startEvent
+   * @param {string=} endEvent
+   * @return {!Promise<ResolverReturnDef>}
    * @private
    */
-  getTimingData_(startEvent, endEvent) {
-    const timingInfo = this.win_['performance']
-        && this.win_['performance']['timing'];
+  getTimingDataAsync_(startEvent, endEvent) {
+    let metric = this.getTimingDataSync_(startEvent, endEvent);
```

Comment:
To clarify: let's keep `loadPromise` itself. But please respond to the original question from @jridgewell